compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -649,29 +649,335 @@ abstract class GenJSCode extends plugins.PluginComponent
         constructorTrees: List[DefDef]): js.Tree = {
       implicit val pos = classSym.pos
 
-      val (primaryCtorTree :: Nil, secondaryCtorTrees) =
-        constructorTrees.partition(_.symbol.isPrimaryConstructor)
-
       // Implementation restriction
-      val sym = primaryCtorTree.symbol
+      val syms = constructorTrees.map(_.symbol)
       val hasBadParam = enteringPhase(currentRun.uncurryPhase) {
-        sym.paramss.flatten.exists(p => p.hasDefault || isRepeated(p))
+        syms.exists(_.paramss.flatten.exists(p => p.hasDefault))
       }
       if (hasBadParam) {
         reporter.error(pos,
             "Implementation restriction: the constructor of a " +
-            "Scala.js-defined JS classes cannot have default parameters nor " +
-            "repeated parameters.")
+            "Scala.js-defined JS classes cannot have default parameters.")
```

Comment:
I belive that it had something to do with the way the default parameter values are defined for constructors. I will have to check.